builtin/credential/cert/path_login.go

```diff
@@ -749,33 +769,78 @@ func (b *backend) loadTrustedCerts(ctx context.Context, storage logical.Storage,
 			conf.OcspThisUpdateMaxAge = entry.OcspThisUpdateMaxAge
 			conf.OcspMaxRetries = entry.OcspMaxRetries
 
-			if len(entry.OcspCaCertificates) > 0 {
-				certs, err := certutil.ParseCertsPEM([]byte(entry.OcspCaCertificates))
-				if err != nil {
-					b.Logger().Error("failed to parse ocsp_ca_certificates", "name", name, "error", err)
-					continue
-				}
-				conf.ExtraCas = certs
+			if len(osp_ca_certs) > 0 {
+				conf.ExtraCas = osp_ca_certs
 			}
 		}
 	}
 
 	if !b.trustedCacheDisabled.Load() {
-		entry := &trusted{
-			pool:          pool,
-			trusted:       trustedCerts,
-			trustedNonCAs: trustedNonCAs,
-			ocspConf:      conf,
+		if anyErrors {
+			// If something went wrong then we are going to set up for an
+			// exponential backoff on reloading the certificates.
+			if cache.retry == nil {
+				cache.retry = &trustedRetry{}
+			}
+
+			// Limits are arbitrary. Max of 2^55 backoff just so that delay
+			// and the the jitter fits into a double. Is that silly? Yes.
+			// 2^55 seconds is more years than the computer will keep
+			// running.
+			if cache.retry.attempt < 55 {
+				cache.retry.attempt += 1
+			}
+
+			d := 1 << cache.retry.attempt
+			pct := (rand.Float64() * 2.0) - 1.0      // between -100% and +100%
+			d += int(math.Floor(float64(d/4) * pct)) // between -25% and +25%
+			cache.retry.deadline = time.Now().Add(time.Duration(d) * time.Second)
+		} else {
+			// No problems, cache is complete, no need to retry.
+			cache.retry = nil
 		}
+
+		cache.trustedNonCAs = trustedNonCAs
+		cache.trusted = trustedCerts
 		if certName == "" {
-			b.trustedCacheFull.Store(entry)
+			b.trustedCacheFull.Store(cache)
 		} else {
-			b.trustedCache.Add(certName, entry)
+			b.trustedCache.Add(certName, cache)
 		}
 	}
 	return
 }
 
+func (b *backend) loadTrustedCert(ctx context.Context, storage logical.Storage, name string) (*CertEntry, []*x509.Certificate, []*x509.Certificate) {
+	entry, err := b.Cert(ctx, storage, strings.TrimPrefix(name, trustedCertPath))
+	if err != nil {
+		b.Logger().Error("failed to load trusted cert", "name", name, "error", err)
+		return nil, nil, nil
+	}
+
+	if entry == nil {
+		// This could happen when the certName was provided and the cert doesn't exist,
+		// or just if between the LIST and the GET the cert was deleted.
+		return nil, nil, nil
```

Comment:
Should this log now? It was previously swallowed and skipped over quietly but this will now be treated as an error which we won't have any idea why from the logs.